mantidproject/mantid
# Added function to separately calculate components of dipole transition matrix

**Description of work.**

Separated out the `getDipoleMatrix()` into three seperate functions that calculate the individual x, y and z components of the matrix:
- `getXDipoleMatrixComponent()`
- `getYDipoleMatrixComponent()`
- `getZDipoleMatrixComponent()`

Added unit test to ensure in the future the sum of the three component functions equals the return value of the `getDipoleMatrix()` function.

**To test:**

1) Create crystal field object (e.g `cf = CrystalField.CrystalField('Ce', 'C2v', B20=0.035, B40=-0.012, B43=-0.027, B60=-0.00012, B63=0.0025, B66=0.0068)`)
2) call getDipoleMatrix() function of object and observe that is returns a matrix that is equal to that output by previous versions of mantid
3) confirm `test_matrix_components_equal_to_summed_matrix`  unit test passed asserting equivalence of the sum of component matrix and return value of getDipoleMatrix() function 

Resolves #33431 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
